sim/core/major_cooldown.go

```diff
@@ -507,15 +507,10 @@ func (mcdm *majorCooldownManager) UpdateMajorCooldowns() {
 
 func (mcdm *majorCooldownManager) sort() {
 	slices.SortStableFunc(mcdm.majorCooldowns, func(m1, m2 *MajorCooldown) int {
-		m1r := m1.ReadyAt()
-		m2r := m2.ReadyAt()
-		if m1r != m2r {
-			return int(m1r - m2r)
+		if r1, r2 := m1.ReadyAt(), m2.ReadyAt(); r1 != r2 {
+			return int(r1 - r2)
 		}
-		if m1.Priority > m2.Priority {
-			return -1
-		}
-		return 1
+		return int(m2.Priority - m1.Priority)
```

Comment:
so this does actually change tests... dunno if its worth caring about

Reply:
hmm, it didn't for me, but most likely could, since it now returns 0 for equal priorities (which is "more correct", in a way).
none of these "x1 - x2" things are "strictly correct", since they fail when "x1 - x2" overflows (e.g. MaxInt32 - (-MaxInt32) or stuff like that), but it won't really matter. currently, using cmp.Compare(x1, x2) is (microbenchmarky, but still) slower, but handles even the NaN-cases in a predictable way.

Reply:
Right, which is what I was trying to summarize haha

Reply:
And you do have test changes in this pr